java/client/src/main/java/glide/api/models/commands/RangeOptions.java

```diff
@@ -0,0 +1,257 @@
+ * glide.api.commands.SortedSetBaseCommands#zrangeWithScores}
+ *
+ * @see <a href="https://redis.io/commands/zrange/">redis.io</a>
+ */
+public class RangeOptions {
+
+    public interface ScoreRange {
+        String toArgs();
+    }
+
+    /** Enumeration representing numeric positive and negative infinity bounds for a sorted set. */
+    @RequiredArgsConstructor
+    public enum InfScoreBound implements ScoreRange {
+        POSITIVE_INFINITY("+inf"),
+        NEGATIVE_INFINITY("-inf");
+
+        private final String redisApi;
+
+        public String toArgs() {
+            return redisApi;
+        }
+    }
+
+    /** Represents a specific numeric score boundary in a sorted set. */
+    public static class ScoreBoundary implements ScoreRange {
+        /** The score value. */
+        private final double bound;
+
+        /** Whether the score value is inclusive. Defaults to true if not set. */
+        private final boolean isInclusive;
+
+        public ScoreBoundary(double bound, boolean isInclusive) {
+            this.bound = bound;
+            this.isInclusive = isInclusive;
+        }
+
+        public ScoreBoundary(double bound) {
+            this.bound = bound;
+            this.isInclusive = true;
+        }
+
+        /** Convert the score boundary to the Redis protocol format. */
+        public String toArgs() {
+            return this.isInclusive ? String.valueOf(this.bound) : "(" + this.bound;
+        }
+    }
+
+    public interface LexRange {
+        String toArgs();
+    }
+
+    /**
+     * Enumeration representing lexicographic positive and negative infinity bounds for sorted set.
+     */
+    @RequiredArgsConstructor
+    public enum InfLexBound implements LexRange {
+        POSITIVE_INFINITY("+"),
+        NEGATIVE_INFINITY("-");
+
+        private final String redisApi;
+
+        @Override
+        public String toArgs() {
+            return redisApi;
+        }
+    }
+
+    /** Represents a specific lexicographic boundary in a sorted set. */
+    public static class LexBoundary implements LexRange {
+        /** The lex value. */
+        private final String value;
+
+        /** Whether the lex value is inclusive. Defaults to true if not set. */
+        private final boolean isInclusive;
+
+        public LexBoundary(@NonNull String value, boolean isInclusive) {
+            this.value = value;
+            this.isInclusive = isInclusive;
+        }
+
+        public LexBoundary(@NonNull String value) {
+            this.value = value;
+            this.isInclusive = true;
+        }
+
+        /** Convert the lex boundary to the Redis protocol format. */
+        @Override
+        public String toArgs() {
+            return this.isInclusive ? "[" + this.value : "(" + this.value;
+        }
+    }
+
+    /**
+     * Represents a limit argument for a range query in a sorted set to be used in <a
+     * href="https://redis.io/commands/zrange">ZRANGE</a> command.<br>
```

Comment:
use `@link`

Reply:
@link to what? Copied this from Node.

Reply:
to `zrange`. use `@link` instead of `a`

Reply:
I just removed it I don't think there is a need to specify a specific command because I believe there are more sorted set commands that will use it.